sim/firesim-lib/src/main/cc/bridges/tracerv.cc

```diff
@@ -106,30 +157,37 @@ void tracerv_t::tick() {
     alignas(4096) uint64_t OUTBUF[QUEUE_DEPTH * 8];
 
     if (outfull) {
-        int can_write = cur_cycle >= start_cycle && cur_cycle < end_cycle;
-
         // TODO. as opt can mmap file and just load directly into it.
         pull(dma_addr, (char*)OUTBUF, QUEUE_DEPTH * 64);
-        if (this->tracefile && can_write) {
+        // TODO: right now, we assume that if one tracefile is set, all are set.
+        if (this->tracefiles[0]) {
```

Comment:
What are the cases under which this would not be true?

Reply:
If there is a problem with the output file? (I was not trying to clean up old code here, but rather adjust it to this use-case)

Reply:
You should either remove this or be consistent for all files.